core/offchain/src/api.rs

```diff
@@ -158,6 +170,32 @@ impl<Storage, KeyProvider> OffchainExt for Api<Storage, KeyProvider> where
 		Ok(CryptoKeyId(id))
 	}
 
+	fn local_authority_pubkey(&self, kind: CryptoKind) -> Result<Vec<u8>, ()> {
+		match self.read_key(None, kind) {
+			Ok(key) => {
+				let public = match key {
+					Key::Sr25519(pair) => pair.public().encode(),
+					Key::Ed25519(pair) => pair.public().encode(),
+				};
+
+				Ok(public)
+			},
+			Err(e) => Err(e),
+		}
+	}
+
+	fn local_network_state(&self) -> Result<Vec<u8>, ()> {
+		let network_state = &self.network_state;
+		let state = LocalNetworkState {
+			external_addresses: network_state.external_addresses(),
+
+			// TODO derive `Serialize` for `libp2p::core::PeerId`
+			peer_id: network_state.peer_id().to_string(),
```

Comment:
```suggestion
			peer_id: network_state.peer_id().to_base58(),
```

`PeerId` is in this weird spot where the serialization option is not obvious, so you have to explicitly call `to_base58()` if you want base58, or `as_bytes()` if you want bytes.